codec-http2/src/main/java/io/netty/handler/codec/http2/Http2FlowController.java

```diff
@@ -22,6 +22,12 @@
 public interface Http2FlowController {
 
     /**
+     * Gets the {@link Http2Connection.PropertyKey} used for accessing the {@link Http2Stream.FlowControlState} property
+     * that this controller associates with each {@link Http2Stream}.
+     */
+    Http2Connection.PropertyKey stateKey();
```

Comment:
@nmittler - Forgot about this one in our talk :) The current code doesn't require properties for flow controller state but instead has 2 properties to store this information.  I agree that removing the setters is a good thing, but this interface (as @buchgr) makes me :cry:.  The "static" properties would be able to resolve this where instead of requiring a "property key" method on the "Http2FlowController"...if you needed to interact with the flow controller state (like the flow controllers do) you would go to the "static property key area" and get the appropriate key.